src/parsers/perf/perfparser.cpp

```diff
@@ -1303,6 +1311,9 @@ void PerfParser::startParseFile(const QString& path, const QString& sysroot, con
     m_bottomUpResults = {};
     m_callerCalleeResults = {};
     m_events = {};
+    m_disassemblyResult = {path, appPath, extraLibPaths, arch};
```

Comment:
m_arch here

Reply:
or even better:

`arch.isEmpty() ? m_arch : arch`

Reply:
There is no m_arch in this class. It is within PerfParser class method. But m_arch is field of PerfParserPrivate. It is computed in void setFeatures(const FeaturesDefinition& features) method of PerfParserPrivate

Reply:
ah, now I see - the arch here is the arg passed in. then just remove the two new lines below, they make no sense. you are setting the arch already, so setting it again when not empty (to itself) makes no sense?!

Reply:
Ok, I will at next iteration